### ISSUE TITLE
Double click on a-href causes full page reload

### ISSUE THREAD

**Issue by thbley** [pinecone-router/router#24]
**Describe the bug**
double click on a-href causes full page reload

**To Reproduce**
Open https://pinecone-example.vercel.app/
Double click "x-handler examples"
Check network console to verify full page reload

**Expected behavior**
Maybe also bind on dblclick in index.ts :: interceptLinks() ?

**Desktop (please complete the following information):**
 - Ubuntu
 - Firefox v121

**Additional context**

**Comment by rehhouari** [comment]
I can't reproduce this on chrome it might be a firefox only issue I'll install it and let you know!

Also, I don't know about binding double click because it might be used for other stuff, in like web apps which are made by Alpine.js.
I don't think it's standard too: [Preact Router](https://github.com/preactjs/preact-router/blob/main/src/index.js#L271) & [Page.js](https://github.com/visionmedia/page.js/blob/master/page.js#L423) only handle click event

**Comment by rehhouari** [comment]
I just installed Firefox and I can't get it to replicate it there too. The fact you're on Ubuntu while I'm on Windows 10 shouldn't be a problem neither.

Only thing that happens when I double click the x-handler button is that it accidentally hit the documentation button in the next page and it opens it in a new tab. The page travel happens in the instant the first click happens.

**Comment by thbley** [comment]
here is a video:
https://github.com/pinecone-router/router/assets/941223/644d21b2-c323-4904-afbf-f4efc579eb68

and some test code:
```
<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="icon" href="data:;base64,iVBORw0KGgo=">
    <script src="https://cdn.jsdelivr.net/npm/pinecone-router@4.x.x/dist/router.min.js"></script>
	<script defer src="https://unpkg.com/alpinejs@3.x.x/dist/cdn.min.js"></script>
</head>

<body>
    <div x-data>
        <a href="./hello/name">Tasks</a>

        <template x-route="./hello/:name">
            <div>
                Hello <span x-text="$router.params.name"></span>
            </div>
        </template>
    </div>
</body>
</html>
```

**Comment by rehhouari** [comment]
I could reproduce the problem with the code you provided, I wonder why the example doesn't have the same issue..

Anyway, as a workaround for now you could use buttons:
```

**Comment by rehhouari** [comment]
I figured out the problem, I'll update with a fix that will handle double clicks

**Comment by rehhouari** [comment]
Fixed in v4.0.2, thanks for opening this! If the problem still persist feel free to reopen

**Comment by thbley** [comment]
I successfully verified the fix with v4.0.2, thank you!
